File: src/main/java/com/mann/mandir/constants/enums/ContentDomain.java
package com.mann.mandir.constants.enums;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum ContentDomain {

    GOD("god", "Devotional content"),
    SCRIPTURES("scriptures", "Hindu scriptures"),
    VEDAS("vedas", "Vedic literature"),
    SEARCH("search", "Cross-domain search");

    private final String path;
    private final String description;

    public static ContentDomain fromString(String value) {
        if (value == null || value.isBlank()) {
            return null;
        }
        for (ContentDomain domain : values()) {
            if (domain.path.equalsIgnoreCase(value) || domain.name().equalsIgnoreCase(value)) {
                return domain;
            }
        }
        throw new IllegalArgumentException("Unknown domain: " + value);
    }
}

File: core/content_domain.py
from __future__ import annotations

from enum import Enum
from typing import Optional


class ContentDomain(str, Enum):
    GOD = "god"
    SCRIPTURES = "scriptures"
    VEDAS = "vedas"
    SEARCH = "search"


def parse_content_domain(value: Optional[str]) -> Optional[ContentDomain]:
    if value is None or not str(value).strip():
        return None
    v = value.strip().lower()
    for d in ContentDomain:
        if d.value == v or d.name.lower() == v:
            return d
    raise ValueError(f"Unknown domain: {value}")
